### PR TITLE
interfaces/mount: add function for parsing fstab-like file

### PR DESCRIPTION
**Pull request by zyga**
This is stacked on top of #3102. The new thing is the 2nd commit with a function for parsing a fstab-like file.

### CODE REVIEW

File: interfaces/mount/entry.go [hunk 1]
```diff
@@ -141,3 +143,29 @@ func ParseEntry(s string) (Entry, error) {
...
+
+// LoadFSTab reads and parses an fstab-like file.
+//
+// The supported format is described by fstab(5).
+func LoadFSTab(reader io.Reader) ([]Entry, error) {
+	var entries []Entry
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		s := strings.TrimSpace(scanner.Text())
+		if i := strings.Index(s, "#"); i != -1 {
```
Comment (chipaca) [inline interfaces/mount/entry.go]: suggest using `IndexByte` directly

File: interfaces/mount/entry.go [hunk 2]
```diff
@@ -141,3 +143,29 @@ func ParseEntry(s string) (Entry, error) {
...
+			s = strings.TrimSpace(s[0:i])
+		}
```
Comment (chipaca) [inline interfaces/mount/entry.go]: to me at least it'd make more sense if you trimmed the comment first, and trimmed the space second. Like
```go
s := scanner.Text()
if i := strings.IndexByte(s, '#') {
    s = s[:i]
}
s = strings.TrimSpace(s)
```

File: interfaces/mount/entry.go (hunk 1 above)
Comment (zyga) [inline interfaces/mount/entry.go]: Nice simplification, I didn't know about `IndexByte`. Done.

File: interfaces/mount/entry.go (hunk 2 above)
Comment (zyga) [inline interfaces/mount/entry.go]: Applied.

**Comment by morphis** [comment]
LGTM

**Pull request merged**